### ISSUE TITLE
Matrix-Notification does not work

### ISSUE THREAD

**Issue by sholl** [crazy-max/diun#297]
## Behaviour

The configured matrix-notification does not work atm in v4.13.0

### Steps to reproduce this issue

1. setup a new matrix user at app.element.io
2. give username and simple password
3. configuration with Env_Vars

### Expected behaviour

> Matrix-notication should be sent to the given room.
The given matrix-user can log into element-Web-client just fine.

### Actual behaviour

> see error logs below

## Configuration

* Diun version : 4.13.0
* Platform (windows/linux) : linux docker-container-setup
* System info (type `uname -a`) : 

> Paste your configuration files here
> * docker-compose.yml
```version: "3.5"

services:
  diun:
    image: crazymax/diun:latest
    container_name: diun
    volumes:
      - "./diun-data:/data"
      - "./userfile.txt:/userfile.txt:ro"
      - "./pwfile.txt:/pwfile.txt:ro"
      - "/var/run/docker.sock:/var/run/docker.sock"
    environment:
      - "TZ=Europe/Paris"
      - "LOG_LEVEL=info"
      - "LOG_JSON=false"
      - "DIUN_DB_PATH=/data/diun.db"
      - "DIUN_WATCH_WORKERS=20"
      - "DIUN_WATCH_SCHEDULE=0 */6 * * *"
      - "DIUN_PROVIDERS_DOCKER=true"
      - "DIUN_PROVIDERS_DOCKER_WATCHSTOPPED=true"
      - "DIUN_NOTIF_MATRIX_HOMESERVERURL=https://matrix-client.matrix.org"
      - "DIUN_NOTIF_MATRIX_USERFILE=/userfile.txt"
      - "DIUN_NOTIF_MATRIX_PASSWORDFILE=/pwfile.txt"
      - "DIUN_NOTIF_MATRIX_ROOMID=!<roomidishere>:matrix.org"
      - "DIUN_NOTIF_MATRIX_MSGTYPE=notice"
    labels:
      - "diun.enable=true"
      - "diun.watch_repo=true"
    restart: always
```

### Logs

```
Tue, 09 Mar 2021 11:52:20 CET INF Starting Diun version=4.13.0
Tue, 09 Mar 2021 11:52:20 CET DBG No configuration file found
Tue, 09 Mar 2021 11:52:20 CET INF Configuration loaded from 10 environment variable(s)
Tue, 09 Mar 2021 11:52:20 CET DBG {
  "db": {
    "path": "/data/diun.db"
  },
  "watch": {
    "workers": 20,
    "schedule": "0 */6 * * *",
    "firstCheckNotif": false,
    "compareDigest": true
  },
  "notif": {
    "matrix": {
      "homeserverURL": "https://matrix-client.matrix.org",
      "userFile": "/userfile.txt",
      "passwordFile": "/pwfile.txt",
      "roomID": "!<roomidishere>:matrix.org",
      "msgType": "notice"
    }
  },
  "providers": {
    "docker": {
      "tlsVerify": true,
      "watchByDefault": false,
      "watchStopped": true
    }
  }
}
Tue, 09 Mar 2021 11:52:20 CET DBG 1 notifier(s) created
Tue, 09 Mar 2021 11:52:20 CET INF Testing notification settings...
Tue, 09 Mar 2021 11:52:20 CET DBG Sending matrix notification... image=docker.io/diun/testnotif:latest
Tue, 09 Mar 2021 11:52:25 CET ERR Cannot logout error="contents=[123 34 101 114 114 99 111 100 101 34 58 34 77 95 77 73 83 83 73 78 71 95 84 79 75 69 78 34 44 34 101 114 114 111 114 34 58 34 77 105 115 115 105 110 103 32 97 99 99 101 115 115 32 116 111 107 101 110 34 125] msg=Failed to POST JSON to /_matrix/client/r0/logout code=401 wrapped=M_MISSING_TOKEN: Missing access token"
Tue, 09 Mar 2021 11:52:25 CET ERR Matrix notification failed error="failed to authenticate Matrix user: contents=[123 34 101 114 114 99 111 100 101 34 58 34 77 95 70 79 82 66 73 68 68 69 78 34 44 34 101 114 114 111 114 34 58 34 73 110 118 97 108 105 100 32 112 97 115 115 119 111 114 100 34 125] msg=Failed to POST JSON to /_matrix/client/r0/login code=403 wrapped=M_FORBIDDEN: Invalid password" image=docker.io/diun/testnotif:latest
/ # 


```

**Comment by crazy-max** [comment]
@sholl 

> `/_matrix/client/r0/login code=403 wrapped=M_FORBIDDEN: Invalid password`

Looks like your credentials are incorrect. Are you sure about the endpoint `https://matrix-client.matrix.org`? Tested myself with `https://matrix.org` and works fine.

**Comment by sholl** [comment]
@crazy-max 

thanks for your feedback. I should have been more precise in my initial report. I woks when providing
DIUN_NOTIF_MATRIX_USER= and DIUN_NOTIF_MATRIX_PASSWORD= as env-vars, but when excluding them to files and using DIUN_NOTIF_MATRIX_USERFILE= and DIUN_NOTIF_MATRIX_PASSWORDFILE= it fails with the above errror message.

I also tried without the :ro-mount-option with no success ether.

Maybe you can reproduce the behaviour as well?

**Comment by crazy-max** [comment]
@sholl Maybe the same issue as #300. Check if your file does not contain a newline ending. To be safe you should create your file like this: `printf "myusername" > ./userfile.txt`

**Comment by sholl** [comment]
@crazy-max, Thanks, I will try your solution and give feedback soon.

**Comment by sholl** [comment]
just verified, using the ```printf``` method for creating userfile and passwordfile resolves the issue, as suggested in #300-

It would be nice documenting this behaviour somewhere. Anyway, thanks for this helpful tool though.
